ts/hooks/useOpenDeepLink.ts

```diff
@@ -0,0 +1,32 @@
+import { useLinkTo } from "@react-navigation/native";
+import * as O from "fp-ts/lib/Option";
+import { pipe } from "fp-ts/lib/function";
+import {
+  IO_FIMS_LINK_PREFIX,
+  IO_INTERNAL_LINK_PREFIX,
+  IO_UNIVERSAL_LINK_PREFIX
+} from "../utils/navigation";
+import { extractPathFromURL } from "../utils/url";
+
+/**
+ * This hook handles deep links. It removes the prefix and navigates to the path using the linkTo function
+ * @returns a function that takes a url and navigates to the path
+ */
+export const useOpenDeepLink = () => {
+  const linkTo = useLinkTo();
+
+  return (url: string) =>
+    pipe(
+      extractPathFromURL(
```

Comment:
wouldn't it be better to handle this with `getInternalRoute` ?

Reply:
Yes it would, but I don't like to use something exported from a component with a specific use case (`Markdown`). Maybe we could move `internalLink.ts` from `ui\Markdown\handlers` to somewhere like `utils`. What do you think?

Reply:
Yes absolutely! let's move into a util

Reply:
Here we go https://github.com/pagopa/io-app/pull/4718/commits/aadcacf2315ce1dbb6c44cd75f22518aca672bed